transaction.go

```diff
@@ -290,6 +293,91 @@ func (t *Transaction) SignalUpdatesApplied() error {
 	return nil
 }
 
+// append is a helper function to append updates to a transaction on which
+// SignalSetupComplete hasn't been called yet
+func (t *Transaction) append(updates []Update, done chan error) {
+	defer close(done)
+
+	// If there is nothing to append we are done
+	if len(updates) == 0 {
+		return
+	}
+
+	// Make sure that the initialization finished
+	<-t.initComplete
+	if t.initErr != nil {
+		done <- t.initErr
+		return
+	}
+
+	// Marshal the data
+	data, err := marshalUpdates(updates)
+	if err != nil {
+		done <- errors.Extend(errors.New("failed to marshal new updates"), err)
+		return
+	}
```

Comment:
an error really shouldn't be possible here -- `marshalUpdates` shouldn't return one. It writes to a `bytes.Buffer`, which [never returns an error](https://golang.org/pkg/bytes/#Buffer.Write).